components/FlexibleHeader/src/MDCFlexibleHeaderView.m

```diff
@@ -298,16 +303,22 @@ - (void)commonMDCFlexibleHeaderViewInit {
   _headerContentImportance = MDCFlexibleHeaderContentImportanceDefault;
   _statusBarHintCanOverlapHeader = YES;
 
-  _minimumHeight = kFlexibleHeaderDefaultHeight + kPreIOS11ExpectedStatusBarHeight;
+  _minimumHeight = kFlexibleHeaderDefaultHeight;
+  _maximumHeight = _minimumHeight;
+  _minHeightIncludingSafeArea = _minimumHeight + kPreIOS11ExpectedStatusBarHeight;
 #if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
   if (@available(iOS 11.0, *)) {
-    // Starting from iOS 11, we should adapt the size of this component to take into account
-    // the new status bar sizes and the "no status bar in landscape" functionality.
-    CGFloat statusBarHeight = [UIApplication mdc_safeSharedApplication].statusBarFrame.size.height;
-    _minimumHeight = kFlexibleHeaderDefaultHeight + statusBarHeight;
+    // Starting from iOS 11, the default behavior of this component is to adapt its height to
+    // account for changes in the Safe Area insets. We're using the status bar frame instead of
+    // the safeAreaInsets property because at this point this view's Safe Area hasn't been
+    // calculated by the OS.
```

Comment:
can we wait until we get added the the window before we do this calculation?
update the value in didMoveToSuperview?

I am concerned that the AppBar is going to be added to a view controller that is not full screen.

Reply:
That's a valid concern, but an issue that precedes iOS 11 and not really related to the iPhone X (previously the status bar height was hardcoded and included in the height). We should fix it though, but I'd prefer to address it on a separate PR.

Reply:
sgtm